File: LeetCode/May challenge/Day 25/Day_25.py
class Solution:
    def maxUncrossedLines(self, A: List[int], B: List[int]) -> int:
        row, col = len(A), len(B)
        
        line_count = [[0 for _ in range(col + 1)] for x in range(row + 1)]
        
        for i in range(1, row + 1):
            for j in range(1, col + 1):
                if A[i - 1] == B[j - 1]:
                    line_count[i][j] = line_count[i - 1][j - 1] + 1
                else:
                    line_count[i][j] = max(line_count[i - 1][j], line_count[i][j - 1])
                
        return line_count[row][col]

File: LeetCode/May challenge/Day 25/Day_25.cpp
class Solution {
public:
    int maxUncrossedLines(vector<int>& A, vector<int>& B) {
        if(A.size()==0 || B.size()==0)
            return 0;
        int res=INT_MIN;
        vector<vector<int>> dp(A.size()+1 ,vector<int>(B.size()+1,0));
        
        for(int i=1;i<=A.size();i++)
        {
            for(int j=1;j<=B.size();j++)
            {
                if(A[i-1]==B[j-1])
                    dp[i][j]=dp[i-1][j-1]+1;
                else
                    dp[i][j]=max(dp[i-1][j],dp[i][j-1]);
                res=max(res,dp[i][j]);
            }
        }
        
        return res;
    }
};
